unstructured/partition/docx.py

```diff
@@ -41,89 +54,93 @@
+    metadata_last_modified: Optional[str] = None,
+) -> List[Element]:
+    """Converts a document to DOCX and then partitions it using partition_docx.
+
+    Works with any file format support by pandoc.
+
+    Parameters
+    ----------
+    source_format
+        The format of the source document, .e.g. odt
+    filename
+        A string defining the target filename path.
+    file
+        A file-like object using "rb" mode --> open(filename, "rb").
+    include_metadata
+        Determines whether or not metadata is included in the metadata attribute on the elements in
+        the output.
     """
+    if "pypandoc" not in globals():
+        raise ImportError("package 'pypandoc' required for this operation but not installed")
 
-    # Recursively get runs.
-    def _get_runs(node, parent):
-        for child in node:
-            # If the child is a run, yield a Run object
-            if child.tag == qn("w:r"):
-                yield Run(child, parent)
-            # If the child is a hyperlink, search for runs within it recursively
-            if child.tag == qn("w:hyperlink"):
-                yield from _get_runs(child, parent)
+    exactly_one(filename=filename, file=file)
 
-    return list(_get_runs(paragraph._element, paragraph))
+    def validate_filename(filename: str) -> str:
+        if not os.path.exists(filename):
+            raise ValueError(f"The file {filename} does not exist.")
+        _, filename_no_path = os.path.split(os.path.abspath(filename))
+        return filename_no_path
+
+    def write_to_tempfile(file: BinaryIO) -> str:
+        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+            tmp.write(file.read())
+            _, filename_no_path = os.path.split(os.path.abspath(tmp.name))
+        return filename_no_path
 
+    filename_no_path = (
+        validate_filename(filename) if filename else write_to_tempfile(cast(BinaryIO, file))
+    )
+
+    base_filename, _ = os.path.splitext(filename_no_path)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        docx_filename = os.path.join(tmpdir, f"{base_filename}.docx")
+        pypandoc.convert_file(  # pyright: ignore
+            filename,
+            "docx",
+            format=source_format,
+            outputfile=docx_filename,
+        )
+        elements = partition_docx(
+            filename=docx_filename,
+            metadata_filename=metadata_filename,
+            include_metadata=include_metadata,
+            metadata_last_modified=metadata_last_modified,
+        )
 
-# Add the runs property to the Paragraph class
-Paragraph.runs = property(lambda self: _get_paragraph_runs(self))
+    return elements
 
 
 @process_metadata()
 @add_metadata_with_filetype(FileType.DOCX)
 @add_chunking_strategy()
 def partition_docx(
     filename: Optional[str] = None,
-    file: Optional[Union[IO[bytes], SpooledTemporaryFile]] = None,
+    file: Optional[Union[BinaryIO, SpooledTemporaryFile[bytes]]] = None,
```

Comment:
I tried to run this locally and encountered an error. Or is it just me?

```
file: Optional[Union[BinaryIO, SpooledTemporaryFile[bytes]]] = None,
TypeError: 'type' object is not subscriptable
```

Reply:
@Klaijan Hmm, good catch. What Python version are you running? I think this problem only arises in 3.7 and 3.8 which would explain why I didn't see it myself. Fix is easy though, just adding:
```python
from __future__ import annotations
```
should do the trick as long as we're support 3.7 and 3.8.

Reply:
Ok, yeah, there are a few problems like this when running under 3.7. I'll get those fixed up. Thanks @Klaijan :)